bazel/execution/service.go

```diff
@@ -181,25 +186,65 @@ func validateExecRequest(req *remoteexecution.ExecuteRequest) error {
 	return nil
 }
 
-// Stub of GetOperation API - most fields omitted but returns a valid hardcoded response.
+// Takes a GetOperation request and forms an ExecuteResponse that is returned as part of a
+// google LongRunning Operation message
 func (s *executionServer) GetOperation(
-	ctx context.Context,
+	_ context.Context,
 	req *longrunning.GetOperationRequest) (*longrunning.Operation, error) {
 	log.Infof("Received GetOperation request: %s", req)
 
 	if !s.IsInitialized() {
 		return nil, status.Error(codes.Internal, "Server not initialized")
 	}
 
-	// TODO get status of job identified by req.Name
+	// TODO(rcouto): This skips over the stats handler increment in github.com/twitter/scoot/scootapi/server/server.go
+	// Do we want to increment that counter? A different counter? Both?
+	js, err := api.GetJobStatus(req.Name, s.sagaCoord)
+	if err != nil {
+		return nil, err
+	}
+	log.Info("Received job status %s", js)
 
 	op := longrunning.Operation{}
-	op.Name = req.Name
+	op.Name = fmt.Sprintf("operations/%s", js.ID)
 	op.Done = true
 
+	// There should only be one task per job, so runStatus/status is the
+	// first & last one we encounter when looping through js.TaskData/TaskStatus
+	if len(js.GetTaskData()) > 1 || len(js.GetTaskStatus()) > 1 {
+		return nil, fmt.Errorf(
+			"TaskData and/or TaskStatus of Bazel job status has len > 1. TaskData: %+v. TaskStatus: %+v",
+			js.GetTaskData(), js.GetTaskStatus())
+	}
+	var runStatus *scoot.RunStatus
+	for _, rs := range js.GetTaskData() {
+		// map from taskID to runStatus
+		runStatus = rs
+	}
+
+	if len(js.GetTaskStatus()) > 0 {
+		var status scoot.Status
+		for _, s := range js.TaskStatus {
+			status = s
+		}
+		if status != js.Status {
+			return nil, fmt.Errorf("Mismatch between task Status and job Status: %s vs %s", status, js.Status)
+		}
+	}
+
+	var br *bazelthrift.ActionResult_
+	if runStatus == nil {
+		br = scoot.RunStatus_BazelResult__DEFAULT
+	} else {
+		br = runStatus.GetBazelResult_()
+	}
+	actionResult := bazelapi.MakeActionResultDomainFromThrift(br)
+
 	eom := &remoteexecution.ExecuteOperationMetadata{
-		Stage: remoteexecution.ExecuteOperationMetadata_COMPLETED,
+		Stage: bazelapi.RunStatusToExecuteOperationMetadata_Stage(runStatus),
 	}
```

Comment:
Add Digest of the action here - I think that's the only additional EOM data required. This might be a good candidate for adding a field to the bazelapi wrapper data struct (record once in the Execute function when we initially calculate the ActionDigest and don't recalculate on each GetOperation request), if this can be passed around easily